arelle/plugin/validate/DBA/rules/__init__.py

```diff
@@ -100,11 +100,37 @@ def errorOnRequiredPositiveFact(
         )
 
 
+def findFactsbyContextId(facts: set[ModelFact],
```

Comment:
Given that mutiple facts can share a context ID, perhaps this function name should include `first`?

```suggestion
def findFirstFactbyContextId(facts: set[ModelFact],
```